### PR TITLE
IGAPP-594: Sort by name and prefix

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

"Landkreis sollte immer vor Stadt sein."

I will create another issue for native as it is also inconsistent there.

### CODE REVIEW

File: web/src/routes/landing/components/CitySelector.js [hunk 1]
```diff
@@ -49,8 +49,17 @@ class CitySelector extends React.PureComponent<PropsType> {
     }
   }
 
+  // Landkreis should come before Stadt
+  sort(cities: Array<CityModel>): Array<CityModel> {
+    return cities.sort(
+      (a, b) =>
+        a.sortingName.localeCompare(b.sortingName) || (a.prefix && b.prefix ? a.prefix.localeCompare(b.prefix) : 0)
```
Comment (klinzo) [inline web/src/routes/landing/components/CitySelector.js:56]: This does not handle comparison of prefix and no prefix, right? I.e. what would happen if there would be `Augsburg` and `Stadt Augsburg`. How about
```suggestion
        a.sortingName.localeCompare(b.sortingName) || (a.prefix || '').localeCompare(b.prefix || ''))
```
I think that should handle all cases correctly.

**Review by klinzo — COMMENTED** [review]
So we did not sort at all before? :astonished:

**Review by klinzo — APPROVED** [review]
Tested on firefox.

File: web/src/routes/landing/components/CitySelector.js (hunk 1 above)
Comment (Taggotty) [reply web/src/routes/landing/components/CitySelector.js]: Great idea. Didn't think about this

**Comment by Taggotty** [comment]
> So we did not sort at all before? astonished

At least I didn't find any sorting. Maybe somewhere in the backend 🤷🏼‍♀️

**Review by SteffiStuffel — APPROVED** [review]
Tested on chrome. :+1:

**Pull request merged**